index.js

```diff
@@ -69,9 +68,14 @@ class ClinicBubbleprof extends events.EventEmitter {
       env: Object.assign({}, process.env, customEnv)
     })
 
-    if (this.detectPort) {
-      proc.stdio[3].once('data', data => this.emit('port', Number(data), proc, () => proc.stdio[3].destroy()))
-    }
+    proc.stdio[3].once('data', data => {
+      if (this.detectPort) {
+        this.emit('port', Number(data), proc, () => proc.stdio[3].destroy())
+      } else if (data.toString() === 'source_warning') {
+        this.emit('warning', 'The code is transpiled, bubbleprof does not support source maps yet.')
+        proc.stdio[3].destroy()
```

Comment:
Destroying fd 3 signals to the app that it needs to quit. In the 'port' event, our callback function is called once the --on-port or --autocannon process exits, and the app should exit in response. For the source map warning, that is not the case: we can just keep going, the warning is to inform the user why the graph might be hard to read.